src/sidebar/util/fetch-config.js

```diff
@@ -0,0 +1,79 @@
+'use strict';
+
+const getApiUrl = require('../get-api-url');
+const hostConfig = require('../host-config');
+const postMessageJsonRpc = require('./postmessage-json-rpc');
+
+function ancestors(window_) {
+  if (window_ === window_.top) {
+    return [];
+  }
+
+  // nb. The top window's `parent` is itself!
+  const ancestors = [];
+  do {
+    window_ = window_.parent;
+    ancestors.push(window_);
+  } while (window_ !== window_.top);
+
+  return ancestors;
+}
+
+/**
+ * Fetch client configuration from an ancestor frame.
+ *
+ * @param {string} origin - The origin of the frame to fetch config from.
+ * @param {Window} window_ - Test seam.
+ * @return {Promise<any>}
+ */
+function fetchConfigFromAncestorFrame(origin, window_=window) {
+  const configResponses = [];
+
+  for (let ancestor of ancestors(window_)) {
+    const timeout = 3000;
+    const result = postMessageJsonRpc.call(
+      ancestor, origin, 'requestConfig', timeout
+    );
+    configResponses.push(result);
+  }
```

Comment:
Ok, so we collect all the responses from all the ancestor frames that we messaged (and some of the responses may be rejected or timed out promises)